package/MDAnalysis/analysis/bat.py

```diff
@@ -0,0 +1,422 @@
+containing atoms a, b, and c and another plane containing b, c, and d.
+For a set of torsions that share atoms b and c, one torsion is defined as
+the primary torsion. The others are defined as improper torsions, differences
+between the raw torsion angle and the primary torsion. This definition reduces
+the correlation between the torsion angles.
+
+Each molecule also has six external coordinates that define its translation and
+rotation in space. The three Cartesian coordinates of the first atom are the
+molecule's translational degrees of freedom. Rotational degreres of freedom are
+specified by the axis-angle convention. The rotation axis is a normalized vector
+pointing from the first to second atom. It is described by the polar angle, phi,
+and azimuthal angle, theta. omega is a third angle that describes the rotation
+of the third atom about the axis.
+
+This module was adapted from AlGDock [Minh2020]_.
+
+
+See Also
+--------
+:class:`~MDAnalysis.analysis.dihedrals.Dihedral`
+   class to calculate dihedral angles for a given set of atoms or residues
+:func:`MDAnalysis.lib.distances.calc_dihedrals()`
+   function to calculate dihedral angles from atom positions
+
+
+Example applications
+--------------------
+
+The :class:`~MDAnalysis.analysis.bat.BAT` class defines bond-angle-torsion
+coordinates based on the topology of an atom group and interconverts between
+Cartesian and BAT coordinate systems. For example, we can determine internal
+coordinates for residues 5-10 of adenylate kinase (AdK). The trajectory is
+included within the test data files::
+
+   import MDAnalysis as mda
+   from MDAnalysisTests.datafiles import PSF, DCD
+   import numpy as np
+
+   u = mda.Universe(PSF, DCD)
+
+   # selection of atomgroups
+   selected_residues = u.select_atoms("resid 5-10")
+
+   from MDAnalysis.analysis.bat import BAT
+   R = BAT(selected_residues)
+
+   # Calculate BAT coordinates for a trajectory
+   R.run()
+
+   # Reconstruct Cartesian coordinates from BAT coordinates
+   bat = R.bat[0]
+   XYZ = R.Cartesian(bat)
+
+   # The difference between the original and reconstructed coordinates
+   # should be zero.
+   print(np.sum(np.abs(XYZ - selected_residues.positions)>1E-6))
+
+After R.run(), the coordinates can be accessed with :attr:`R.bat`.
+
+
+References
+----------
+
+.. [Chang2003] Chang, Chia-En, Michael J Potter, and Michael K Gilson (2003).
+   "Calculation of Molecular Configuration Integrals". *Journal of Physical
+   Chemistry B* 107(4): 1048–55. doi:`10.1021/jp027149c
+   <https://doi.org/10.1021/jp027149c>`_
+
+.. [Hikiri2016] Hikiri, Simon, Takashi Yoshidome, and Mitsunori Ikeguchi (2016).
+   "Computational Methods for Configurational Entropy Using Internal and
+   Cartesian Coordinates." *Journal of Chemical Theory and Computation*
+   12(12): 5990–6000. doi:`10.1021/acs.jctc.6b00563
+   <https://doi.org/10.1021/acs.jctc.6b00563>`_
+
+.. [Minh2020] Minh, David D L (2020). "Alchemical Grid Dock (AlGDock): Binding
```

Comment:
if you want you can check out how we use duecredit for citations

Reply:
Hmm, that's an interesting feature. How would I add citations to the duecredit report for those who use the BAT module (or any other module I write)?

(mdanalysis-dev) Minh-IIT-MBP2018:[~/clusters/CCB/FEPiC]: more test_credit.py
import MDAnalysis as mda
from MDAnalysisTests.datafiles import PSF, DCD
u = mda.Universe(PSF, DCD)

# selection of atomgroups
selected_residues = u.select_atoms("resid 5-10")

from MDAnalysis.analysis.bat import BAT

self = BAT(selected_residues)

# Calculates BAT coordinates for a trajectory
self.run()
(mdanalysis-dev) Minh-IIT-MBP2018:[~/clusters/CCB/FEPiC]: python -m duecredit test_credit.py
/Users/dminh/Documents/GitHub/mdanalysis/testsuite/MDAnalysisTests/__init__.py:126: MatplotlibDeprecationWarning: The 'warn' parameter of use() is deprecated since Matplotlib 3.1 and will be removed in 3.3.  If any parameter follows 'warn', they should be pass as keyword, not positionally.
  matplotlib.use('agg', warn=False)

DueCredit Report:
- Molecular simulation analysis library / MDAnalysis (v 0.20.2-dev0) [1, 2]
- Scientific tools library / numpy (v 1.18.1) [3]

2 packages cited
0 modules cited
0 functions cited

References
----------

[1] Gowers, R. et al., 2016. MDAnalysis: A Python Package for the Rapid Analysis of Molecular Dynamics Simulations. In Proceedings of the 15th Python in Science Conference.  SciPy.
[2] Michaud-Agrawal, N. et al., 2011. MDAnalysis: A toolkit for the analysis of molecular dynamics simulations. Journal of Computational Chemistry, 32(10), pp.2319–2327.
[3] Van Der Walt, S., Colbert, S.C. & Varoquaux, G., 2011. The NumPy array: a structure for efficient numerical computation. Computing in Science & Engineering, 13(2), pp.22–30.

Reply:
For the whole module:
```python
from ..due import due, Doi

due.cite(Doi("10.1002/jcc.2603"),
         description="Bond-angle-torsions coordinates",
         path="MDAnalysis.analysis.bat",
         cite_module=True)
del Doi
```

For a method or function
```python
class BAT():
    @due.dcite(Doi("10.1002/jcc.2603"),
            description="Bond-angle-torsions coordinates analysis",
            path="MDAnalysis.analysis.bat.BAT")
     def __init__(....):
        ....
```
(at least I think that's how you decorate the class; see the [duecredit README](https://github.com/duecredit/duecredit/blob/master/README.md) for details)